### ISSUE TITLE
Deprecation Warnings:

### ISSUE THREAD

**Issue by ViolanteCodes** [ButterCMS/laravel-starter-buttercms#2]
On `composer install` command, I'm seeing some deprecation warnings: 

This was immediately after installing PHP 8.0 (so version 8.0.17, which is less than pinned requirement of 8.0.2)
https://share.getcloudapp.com/RBunZZ0m

But then once I'd updated to most recent support version + upgraded dependencies, I got this: 

```
butterenv) maria@maria-Galago-Pro:~/Programming/butter/starters/laravel-starter-buttercms$ composer install
PHP Deprecated:  Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Console/Helper/HelperSet.php on line 104

Deprecated: Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Console/Helper/HelperSet.php on line 104
Deprecation Notice: Optional parameter $schema declared before required parameter $path is implicitly treated as a required parameter in /usr/share/php/JsonSchema/Constraints/UndefinedConstraint.php:62
Deprecation Notice: Optional parameter $schema declared before required parameter $path is implicitly treated as a required parameter in /usr/share/php/JsonSchema/Constraints/UndefinedConstraint.php:108
Deprecation Notice: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /usr/share/php/JsonSchema/Constraints/Constraint.php:48
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Symfony\Component\Process\Process::getIterator($flags = 0) should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Process/Process.php:606
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Composer\Repository\CompositeRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/CompositeRepository.php:139
Loading composer repositories with package information
Deprecation Notice: Return type of Composer\DependencyResolver\Pool::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Pool.php:166
Installing dependencies (including require-dev) from lock file
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSet.php:114
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSet::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSet.php:99
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:155
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:165
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:160
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:170
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:150
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:145
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:36
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:46
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:41
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:89
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:70
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for tijsverkoyen/css-to-inline-styles 2.2.4 -> satisfiable by tijsverkoyen/css-to-inline-styles[2.2.4].
    - tijsverkoyen/css-to-inline-styles 2.2.4 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 2
    - Installation request for phar-io/manifest 2.0.3 -> satisfiable by phar-io/manifest[2.0.3].
    - phar-io/manifest 2.0.3 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 3
    - Installation request for phpunit/php-code-coverage 9.2.15 -> satisfiable by phpunit/php-code-coverage[9.2.15].
    - phpunit/php-code-coverage 9.2.15 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 4
    - Installation request for phpunit/phpunit 9.5.19 -> satisfiable by phpunit/phpunit[9.5.19].
    - phpunit/phpunit 9.5.19 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 5
    - Installation request for spatie/laravel-ignition 1.1.1 -> satisfiable by spatie/laravel-ignition[1.1.1].
    - spatie/laravel-ignition 1.1.1 requires ext-curl * -> the requested PHP extension curl is missing from your system.
  Problem 6
    - Installation request for theseer/tokenizer 1.2.1 -> satisfiable by theseer/tokenizer[1.2.1].
    - theseer/tokenizer 1.2.1 requires ext-dom * -> the requested PHP extension dom is missing from your system.
  Problem 7
    - tijsverkoyen/css-to-inline-styles 2.2.4 requires ext-dom * -> the requested PHP extension dom is missing from your system.
    - laravel/framework v9.5.1 requires tijsverkoyen/css-to-inline-styles ^2.2.2 -> satisfiable by tijsverkoyen/css-to-inline-styles[2.2.4].
    - Installation request for laravel/framework v9.5.1 -> satisfiable by laravel/framework[v9.5.1].

  To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/8.1/cli/php.ini
    - /etc/php/8.1/cli/conf.d/10-opcache.ini
    - /etc/php/8.1/cli/conf.d/10-pdo.ini
    - /etc/php/8.1/cli/conf.d/20-calendar.ini
    - /etc/php/8.1/cli/conf.d/20-ctype.ini
    - /etc/php/8.1/cli/conf.d/20-exif.ini
    - /etc/php/8.1/cli/conf.d/20-ffi.ini
    - /etc/php/8.1/cli/conf.d/20-fileinfo.ini
    - /etc/php/8.1/cli/conf.d/20-ftp.ini
    - /etc/php/8.1/cli/conf.d/20-gettext.ini
    - /etc/php/8.1/cli/conf.d/20-iconv.ini
    - /etc/php/8.1/cli/conf.d/20-mbstring.ini
    - /etc/php/8.1/cli/conf.d/20-phar.ini
    - /etc/php/8.1/cli/conf.d/20-posix.ini
    - /etc/php/8.1/cli/conf.d/20-readline.ini
    - /etc/php/8.1/cli/conf.d/20-shmop.ini
    - /etc/php/8.1/cli/conf.d/20-sockets.ini
    - /etc/php/8.1/cli/conf.d/20-sysvmsg.ini
    - /etc/php/8.1/cli/conf.d/20-sysvsem.ini
    - /etc/php/8.1/cli/conf.d/20-sysvshm.ini
    - /etc/php/8.1/cli/conf.d/20-tokenizer.ini
  You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
```

My apologies if this is all basic info to a veteran PHP developer, but can you walk me through what I'm supposed to do next? Is it expected that all of these warnings are going to resolve if I install the various packages one by one? Is there a way to have composer automatically install these packages? Ideally, we'd like to minimize warnings, etc., as much as possible and have people to be able to run "composer install" and maybe a few commands if necessary from command line and immediately jump into the project.

**Comment by prokopsimek** [comment]
@janondracek1 Hi, please, can you take a look at this error? Thank you

**Comment by janondracek1** [comment]
@ViolanteCodes 
There was a slight issue of `composer.lock` being included, which is overall a bad practice and is known to cause issues when installing across different environments. The mentioned file was generated on my environment of PHP 8.1.4.

I am currently unable to easily replicate your issue, as I do not have a clean Linux installation at my disposal, so I will try and guide you to resolve the mentioned issues.

From what I see in the dump you provided, you are missing vital extensions, that need to be manually configured via the mentioned `php.ini`. As far as I know, this is not possible to be done automatically, as it varies from environment to environment.
To be exact, you are missing `ext-dom` and `ext-curl`, to enable those, you need to go into the folder of your PHP installation and open up `php.ini`.

The `ext-dom` extension is for XML manipulation and `ext-curl` is for cURL, heavily utilized library for HTTP requests via PHP.
`ext-curl` is provided by default, so in your `php.ini` locate `;extension=curl` and remove the `;` character, which indicates a comment line.

As for the `ext-dom`, it needs to be installed separately, look for `php-xml`. Once the installation is complete, try to rerun `composer-install`, it should go through.
If not, I will try to help you further.

**Comment by ViolanteCodes** [comment]
@jakelumetta I wanted to loop you in here to see how you wanted to handle this. 

If there is not way to automatically install these extensions, then we should probably have documentation in the readme (maybe a section at the bottom?) that at least points the user to how to easily resolve the issue or at least confirms that this is expected behavior and that the right step to take is to install the extensions. 

I say this because when confronted with this error myself, although I thought *maybe I should install these* I also thought *or maybe it's broken?*

What do you think about having something in the installation instructions that says, "If you receive this message" (dependency related message), then drop to `this section` below, which links to a quick walkthrough for people who are maybe a bit newer to PHP. That way, it doesn't break up the reading workflow for veteran developers, but it also confirms that these extensions are meant to be installed.

We could have a cleaner version of [this](https://stackoverflow.com/questions/43408604/php7-install-ext-dom-issue/43408658#43408658) maybe for the instructions?

**Comment by ViolanteCodes** [comment]
Update @janondracek1 

after running
`sudo apt install php-xml`
`sudo apt-get install php-curl`

I was able to run `composer install`, however, I'm still seeing a number of deprecation warnings and notices. Is it not possible to resolve these? They're at the top and bottom of this message below.

**PHP Deprecated:  Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Console/Helper/HelperSet.php on line 104

Deprecated: Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Console/Helper/HelperSet.php on line 104
Deprecation Notice: Optional parameter $schema declared before required parameter $path is implicitly treated as a required parameter in /usr/share/php/JsonSchema/Constraints/UndefinedConstraint.php:62
Deprecation Notice: Optional parameter $schema declared before required parameter $path is implicitly treated as a required parameter in /usr/share/php/JsonSchema/Constraints/UndefinedConstraint.php:108
Deprecation Notice: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /usr/share/php/JsonSchema/Constraints/Constraint.php:48
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Symfony\Component\Process\Process::getIterator($flags = 0) should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Process/Process.php:606
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Return type of Composer\Repository\ArrayRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/ArrayRepository.php:206
Deprecation Notice: Return type of Composer\Repository\CompositeRepository::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/Repository/CompositeRepository.php:139
Loading composer repositories with package information
Deprecation Notice: Return type of Composer\DependencyResolver\Pool::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Pool.php:166
Installing dependencies (including require-dev) from lock file
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSet.php:114
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSet::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSet.php:99
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:155
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:165
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:160
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:170
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:150
Deprecation Notice: Return type of Composer\DependencyResolver\Decisions::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/Decisions.php:145
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:36
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:46
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:41
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:89
Deprecation Notice: Return type of Composer\DependencyResolver\RuleSetIterator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Composer/DependencyResolver/RuleSetIterator.php:70
Package operations: 123 installs, 0 updates, 0 removals
  - Installing symfony/deprecation-contracts (v3.0.0): Deprecation Notice: Return type of Symfony\Component\Finder\Finder::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Finder.php:620
Deprecation Notice: Return type of Symfony\Component\Finder\Finder::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Finder.php:691
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::getChildren() should either be compatible with RecursiveDirectoryIterator::getChildren(): RecursiveDirectoryIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php:81
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::rewind() should either be compatible with FilesystemIterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php:109
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::current() should either be compatible with FilesystemIterator::current(): SplFileInfo|FilesystemIterator|string, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php:61
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\DepthRangeFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/DepthRangeFilterIterator.php:41
Downloading (100%)         
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\FilenameFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/FilenameFilterIterator.php:28**
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing psr/http-client (1.0.1): Loading from cache
  - Installing ralouphie/getallheaders (3.0.3): Loading from cache
  - Installing psr/http-factory (1.0.1): Loading from cache
  - Installing guzzlehttp/psr7 (2.2.1): Downloading (100%)         
  - Installing guzzlehttp/promises (1.5.1): Loading from cache
  - Installing guzzlehttp/guzzle (7.4.2): Downloading (100%)         
  - Installing buttercms/buttercms-php (2.4.1): Loading from cache
  - Installing doctrine/inflector (2.0.4): Loading from cache
  - Installing symfony/polyfill-ctype (v1.25.0): Loading from cache
  - Installing webmozart/assert (1.10.0): Loading from cache
  - Installing dragonmantank/cron-expression (v3.3.1): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.25.0): Loading from cache
  - Installing symfony/http-foundation (v6.0.6): Downloading (100%)         
  - Installing fruitcake/php-cors (v1.2.0): Downloading (100%)         
  - Installing voku/portable-ascii (2.0.1): Downloading (100%)         
  - Installing symfony/polyfill-php80 (v1.25.0): Downloading (100%)         
  - Installing phpoption/phpoption (1.8.1): Loading from cache
  - Installing graham-campbell/result-type (v1.0.4): Loading from cache
  - Installing vlucas/phpdotenv (v5.4.1): Loading from cache
  - Installing symfony/css-selector (v6.0.3): Downloading (100%)         
  - Installing tijsverkoyen/css-to-inline-styles (2.2.4): Loading from cache
  - Installing symfony/var-dumper (v6.0.6): Downloading (100%)         
  - Installing symfony/routing (v6.0.5): Downloading (100%)         
  - Installing symfony/process (v6.0.5): Downloading (100%)         
  - Installing symfony/polyfill-php72 (v1.25.0): Loading from cache
  - Installing symfony/polyfill-intl-normalizer (v1.25.0): Loading from cache
  - Installing symfony/polyfill-intl-idn (v1.25.0): Loading from cache
  - Installing symfony/mime (v6.0.3): Downloading (100%)         
  - Installing psr/container (2.0.2): Downloading (100%)         
  - Installing symfony/service-contracts (v3.0.0): Downloading (100%)         
  - Installing psr/event-dispatcher (1.0.0): Loading from cache
  - Installing symfony/event-dispatcher-contracts (v3.0.0): Downloading (100%)         
  - Installing symfony/event-dispatcher (v6.0.3): Downloading (100%)         
  - Installing psr/log (3.0.0): Downloading (100%)         
  - Installing doctrine/lexer (1.2.3): Downloading (100%)         
  - Installing egulias/email-validator (3.1.2): Downloading (100%)         
  - Installing symfony/mailer (v6.0.5): Downloading (100%)         
  - Installing symfony/error-handler (v6.0.3): Downloading (100%)         
  - Installing symfony/http-kernel (v6.0.6): Downloading (100%)         
  - Installing symfony/finder (v6.0.3): Downloading (100%)         
  - Installing symfony/polyfill-intl-grapheme (v1.25.0): Loading from cache
  - Installing symfony/string (v6.0.3): Downloading (100%)         
  - Installing symfony/console (v6.0.5): Downloading (100%)         
  - Installing symfony/polyfill-php81 (v1.25.0): Loading from cache
  - Installing ramsey/collection (1.2.2): Loading from cache
  - Installing brick/math (0.9.3): Loading from cache
  - Installing ramsey/uuid (4.3.1): Downloading (100%)         
  - Installing psr/simple-cache (3.0.0): Downloading (100%)         
  - Installing symfony/translation-contracts (v3.0.0): Downloading (100%)         
  - Installing symfony/translation (v6.0.6): Downloading (100%)         
  - Installing nesbot/carbon (2.57.0): Downloading (100%)         
  - Installing monolog/monolog (2.4.0): Downloading (100%)         
  - Installing league/mime-type-detection (1.9.0): Loading from cache
  - Installing league/flysystem (3.0.12): Downloading (100%)         
  - Installing nette/utils (v3.2.7): Downloading (100%)         
  - Installing nette/schema (v1.2.2): Loading from cache
  - Installing dflydev/dot-access-data (v3.0.1): Loading from cache
  - Installing league/config (v1.1.1): Loading from cache
  - Installing league/commonmark (2.2.3): Downloading (100%)         
  - Installing laravel/serializable-closure (v1.1.1): Downloading (100%)         
  - Installing laravel/framework (v9.5.1): Downloading (100%)         
  - Installing laravel/sanctum (v2.14.2): Downloading (100%)         
  - Installing nikic/php-parser (v4.13.2): Loading from cache
  - Installing psy/psysh (v0.11.2): Downloading (100%)         
  - Installing laravel/tinker (v2.7.1): Downloading (100%)         
  - Installing composer/semver (3.3.1): Downloading (100%)         
  - Installing composer/pcre (3.0.0): Downloading (100%)         
  - Installing composer/xdebug-handler (3.0.3): Downloading (100%)         
  - Installing psr/cache (3.0.0): Downloading (100%)         
  - Installing doctrine/annotations (1.13.2): Downloading (100%)         
  - Installing fakerphp/faker (v1.19.0): Downloading (100%)         
  - Installing hamcrest/hamcrest-php (v2.0.1): Loading from cache
  - Installing laravel/sail (v1.13.7): Downloading (100%)         
  - Installing filp/whoops (2.14.5): Loading from cache
  - Installing facade/ignition-contracts (1.0.2): Loading from cache
  - Installing nunomaduro/collision (v6.1.0): Downloading (100%)         
  - Installing phpstan/phpstan (1.5.2): Downloading (100%)         
  - Installing phpmyadmin/sql-parser (5.5.0): Downloading (100%)         
  - Installing mockery/mockery (1.5.0): Downloading (100%)         
  - Installing nunomaduro/larastan (v2.1.3): Downloading (100%)         
  - Installing php-cs-fixer/diff (v2.0.2): Downloading (100%)         
  - Installing phpdocumentor/reflection-common (2.2.0): Loading from cache
  - Installing phpdocumentor/type-resolver (1.6.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (5.3.0): Loading from cache
  - Installing sebastian/version (3.0.2): Loading from cache
  - Installing sebastian/type (3.0.0): Downloading (100%)         
  - Installing sebastian/resource-operations (3.0.3): Loading from cache
  - Installing sebastian/recursion-context (4.0.4): Loading from cache
  - Installing sebastian/object-reflector (2.0.4): Loading from cache
  - Installing sebastian/object-enumerator (4.0.4): Loading from cache
  - Installing sebastian/global-state (5.0.5): Downloading (100%)         
  - Installing sebastian/exporter (4.0.4): Loading from cache
  - Installing sebastian/environment (5.1.3): Loading from cache
  - Installing sebastian/diff (4.0.4): Loading from cache
  - Installing sebastian/comparator (4.0.6): Loading from cache
  - Installing sebastian/code-unit (1.0.8): Loading from cache
  - Installing sebastian/cli-parser (1.0.1): Loading from cache
  - Installing phpunit/php-timer (5.0.3): Loading from cache
  - Installing phpunit/php-text-template (2.0.4): Loading from cache
  - Installing phpunit/php-invoker (3.1.1): Loading from cache
  - Installing phpunit/php-file-iterator (3.0.6): Loading from cache
  - Installing theseer/tokenizer (1.2.1): Loading from cache
  - Installing sebastian/lines-of-code (1.0.3): Loading from cache
  - Installing sebastian/complexity (2.0.2): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (2.0.3): Loading from cache
  - Installing phpunit/php-code-coverage (9.2.15): Downloading (100%)         
  - Installing doctrine/instantiator (1.4.1): Downloading (100%)         
  - Installing phpspec/prophecy (v1.15.0): Loading from cache
  - Installing phar-io/version (3.2.1): Downloading (100%)         
  - Installing phar-io/manifest (2.0.3): Loading from cache
  - Installing myclabs/deep-copy (1.11.0): Downloading (100%)         
  - Installing phpunit/phpunit (9.5.19): Downloading (100%)         
  - Installing spatie/backtrace (1.2.1): Downloading (100%)         
  - Installing spatie/flare-client-php (1.1.0): Downloading (100%)         
  - Installing spatie/ignition (1.2.7): Downloading (100%)         
  - Installing spatie/laravel-ignition (1.1.1): Downloading (100%)         
  - Installing symfony/stopwatch (v6.0.5): Downloading (100%)         
  - Installing symfony/options-resolver (v6.0.3): Downloading (100%)         
  - Installing symfony/filesystem (v6.0.6): Downloading (100%)         
  - Installing friendsofphp/php-cs-fixer (v3.8.0): Downloading (100%)         
  - Installing stechstudio/laravel-php-cs-fixer (v3.1.0): Downloading (100%)         
guzzlehttp/psr7 suggests installing laminas/laminas-httphandlerrunner (Emit PSR-7 responses)
guzzlehttp/guzzle suggests installing ext-intl (Required for Internationalized Domain Name (IDN) support)
voku/portable-ascii suggests installing ext-intl (Use Intl for transliterator_transliterate() support)
symfony/var-dumper suggests installing ext-intl (To show region name in time zone dump)
symfony/routing suggests installing symfony/config (For using the all-in-one router or any loader)
symfony/routing suggests installing symfony/expression-language (For using expression matching)
symfony/routing suggests installing symfony/yaml (For using the YAML loader)
symfony/polyfill-intl-normalizer suggests installing ext-intl (For best performance)
symfony/polyfill-intl-idn suggests installing ext-intl (For best performance)
symfony/service-contracts suggests installing symfony/service-implementation
symfony/event-dispatcher suggests installing symfony/dependency-injection
egulias/email-validator suggests installing ext-intl (PHP Internationalization Libraries are required to use the SpoofChecking validation)
symfony/http-kernel suggests installing symfony/browser-kit
symfony/http-kernel suggests installing symfony/config
symfony/http-kernel suggests installing symfony/dependency-injection
symfony/polyfill-intl-grapheme suggests installing ext-intl (For best performance)
symfony/console suggests installing symfony/lock
ramsey/uuid suggests installing ext-bcmath (Enables faster math with arbitrary-precision integers using BCMath.)
ramsey/uuid suggests installing ext-gmp (Enables faster math with arbitrary-precision integers using GMP.)
ramsey/uuid suggests installing ext-uuid (Enables the use of PeclUuidTimeGenerator and PeclUuidRandomGenerator.)
ramsey/uuid suggests installing paragonie/random-lib (Provides RandomLib for use with the RandomLibAdapter)
ramsey/uuid suggests installing ramsey/uuid-doctrine (Allows the use of Ramsey\Uuid\Uuid as Doctrine field type.)
symfony/translation suggests installing symfony/config
symfony/translation suggests installing symfony/yaml
monolog/monolog suggests installing aws/aws-sdk-php (Allow sending log messages to AWS services like DynamoDB)
monolog/monolog suggests installing doctrine/couchdb (Allow sending log messages to a CouchDB server)
monolog/monolog suggests installing elasticsearch/elasticsearch (Allow sending log messages to an Elasticsearch server via official client)
monolog/monolog suggests installing ext-amqp (Allow sending log messages to an AMQP server (1.0+ required))
monolog/monolog suggests installing ext-mongodb (Allow sending log messages to a MongoDB server (via driver))
monolog/monolog suggests installing graylog2/gelf-php (Allow sending log messages to a GrayLog2 server)
monolog/monolog suggests installing mongodb/mongodb (Allow sending log messages to a MongoDB server (via library))
monolog/monolog suggests installing php-amqplib/php-amqplib (Allow sending log messages to an AMQP server using php-amqplib)
monolog/monolog suggests installing php-console/php-console (Allow sending log messages to Google Chrome)
monolog/monolog suggests installing rollbar/rollbar (Allow sending log messages to Rollbar)
monolog/monolog suggests installing ruflin/elastica (Allow sending log messages to an Elastic Search server)
nette/utils suggests installing ext-gd (to use Image)
nette/utils suggests installing ext-intl (to use Strings::webalize(), toAscii(), normalize() and compare())
league/commonmark suggests installing symfony/yaml (v2.3+ required if using the Front Matter extension)
laravel/framework suggests installing ably/ably-php (Required to use the Ably broadcast driver (^1.0).)
laravel/framework suggests installing aws/aws-sdk-php (Required to use the SQS queue driver, DynamoDb failed job storage, and SES mail driver (^3.198.1).)
laravel/framework suggests installing brianium/paratest (Required to run tests in parallel (^6.0).)
laravel/framework suggests installing doctrine/dbal (Required to rename columns and drop SQLite columns (^2.13.3|^3.1.4).)
laravel/framework suggests installing ext-bcmath (Required to use the multiple_of validation rule.)
laravel/framework suggests installing ext-gd (Required to use Illuminate\Http\Testing\FileFactory::image().)
laravel/framework suggests installing ext-memcached (Required to use the memcache cache driver.)
laravel/framework suggests installing ext-redis (Required to use the Redis cache and queue drivers (^4.0|^5.0).)
laravel/framework suggests installing league/flysystem-aws-s3-v3 (Required to use the Flysystem S3 driver (^3.0).)
laravel/framework suggests installing league/flysystem-ftp (Required to use the Flysystem FTP driver (^3.0).)
laravel/framework suggests installing league/flysystem-sftp-v3 (Required to use the Flysystem SFTP driver (^3.0).)
laravel/framework suggests installing nyholm/psr7 (Required to use PSR-7 bridging features (^1.2).)
laravel/framework suggests installing pda/pheanstalk (Required to use the beanstalk queue driver (^4.0).)
laravel/framework suggests installing predis/predis (Required to use the predis connector (^1.1.9).)
laravel/framework suggests installing pusher/pusher-php-server (Required to use the Pusher broadcast driver (^6.0|^7.0).)
laravel/framework suggests installing symfony/cache (Required to PSR-6 cache bridge (^6.0).)
laravel/framework suggests installing symfony/http-client (Required to enable support for the Symfony API mail transports (^6.0).)
laravel/framework suggests installing symfony/mailgun-mailer (Required to enable support for the Mailgun mail transport (^6.0).)
laravel/framework suggests installing symfony/postmark-mailer (Required to enable support for the Postmark mail transport (^6.0).)
laravel/framework suggests installing symfony/psr-http-message-bridge (Required to use PSR-7 bridging features (^2.0).)
psy/psysh suggests installing ext-pdo-sqlite (The doc command requires SQLite to work.)
psy/psysh suggests installing hoa/console (A pure PHP readline implementation. You'll want this if your PHP install doesn't already support readline or libedit.)
fakerphp/faker suggests installing doctrine/orm (Required to use Faker\ORM\Doctrine)
filp/whoops suggests installing whoops/soap (Formats errors as SOAP responses)
phpmyadmin/sql-parser suggests installing phpmyadmin/motranslator (Translate messages to your favorite locale)
nunomaduro/larastan suggests installing orchestra/testbench (Using Larastan for analysing a package needs Testbench)
sebastian/global-state suggests installing ext-uopz (*)
phpunit/php-code-coverage suggests installing ext-pcov (*)
phpunit/php-code-coverage suggests installing ext-xdebug (*)
phpunit/phpunit suggests installing ext-soap (*)
phpunit/phpunit suggests installing ext-xdebug (*)
Generating optimized autoload files
**Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\FileTypeFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/FileTypeFilterIterator.php:42
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/ExcludeDirectoryFilterIterator.php:55
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::hasChildren() should either be compatible with RecursiveIterator::hasChildren(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/ExcludeDirectoryFilterIterator.php:71
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::getChildren() should either be compatible with RecursiveIterator::getChildren(): ?RecursiveIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/ExcludeDirectoryFilterIterator.php:76
Deprecation Notice: Return type of Symfony\Component\Finder\Iterator\PathFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/php/Symfony/Component/Finder/Iterator/PathFilterIterator.php:27
Deprecation Notice: preg_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:257
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/EventDispatcher/EventDispatcher.php:322**

**Comment by jakelumetta** [comment]
Why do we need `ext-dom` and `ext-curl`?  Our API returns JSON data and we should not be doing raw curls requests, this project should utilize our PHP SDK to interact with the ButterCMS API: https://github.com/ButterCMS/buttercms-php
 - Also note our PHP SDK utilizes `GuzzleHttp\Client` for HTTP requests.

**Comment by janondracek1** [comment]
@jakelumetta 
This project indeed utilizes the PHP SDK you listed above.

`ext-dom` and `ext-curl` is needed by the dependencies of `laravel/laravel`, as listed in the first provided dump.
I haven't touched dependencies of the starter Laravel project, except for adding the implementation for ButterCMS API and some development dependencies.

**Comment by ViolanteCodes** [comment]
@jakelumetta please let me know if the adjust-the-readme solution works for you, per comments above :) thanks!

**Comment by jakelumetta** [comment]
@janondracek1 Ah ok - if `ext-dom` and `ext-curl` are core to laravel and there's no way around it then I don't think we have much choice but to include them though it would be nice if we were able to exclude them if they are a little tricky to install 🤷‍♂️ 

@ViolanteCodes sure we can add a section to the bottom of the README. I would be curious if other devs on our team run into this issue as well. i.e. is this common?

**Comment by janondracek1** [comment]
@jakelumetta 
I tried trimming the sources as much as I could, without any direct impact and I am not entirely sure about dropping these dependencies. 

@ViolanteCodes 
If any issues arise, should I commit directly to master? Because I created a new PR #3 where I removed the mentioned package.lock, so try switching to that branch and trying running `composer install` if the deprecation warnings still persist.

**Comment by ViolanteCodes** [comment]
@jakelumetta
@janondracek1 

Sorry for a bit of a late reply on this but I've been down the rabbit hole with some reading to try to figure out what was going on here, why I was getting deprecation warnings for the code, etc. 

When I checked out the PR that @janondracek1 posted, between the Fatal Argument errors on `composer install` with `-vvv` flag  and the `Uncaught ArgumentCountError: array_merge() does not accept unknown named parameters in /usr/share/php/Composer/DependencyResolver/DefaultPolicy.php:84`, I was able to chase down a thread [https://github.com/composer/composer/issues/9097](https://github.com/composer/composer/issues/9097) which led me to believe the issue was with my version of composer. 

I tried the `composer self-update` command but was unable to update, which led to this thread [https://stackoverflow.com/questions/37772798/unable-to-self-update-composer](https://stackoverflow.com/questions/37772798/unable-to-self-update-composer) - apparently, the version of composer installed with `apt` on linux is very outdated.

I uninstalled all the packages I'd installed thus far, including `ext-dom` and `ext-curl`, did a `sudo apt remove composer`, and installed composer directly from source [https://getcomposer.org/download/](https://getcomposer.org/download/) as well as set composer on PATH. 

I was then able to check out master branch, run `composer install`, set my token, and successfully run the local server to launch the project, without downloading or installing any extra packages.

**So, here are my conclusions and questions:**
- Personally, I am in favor of having the composer.lock file in master and pinning the dependencies to versions for a working copy as is currently set. @janondracek1, is there a way to pin a version of composer?
- I think that the best approach might be to have a link in the install section that says something like `If you're having problems with installation, chances are good you need to update either your PHP version (this project requires 8.0.2) or your composer version (this project requires composer 2.3.5). Drop to #troubleshooting for more instructions`. 
- My thought process here is that if the user runs the `composer install` command with an outdated version of composer, they'll get led down this rabbit-hole of installing dependency after dependency, deprecation warnings, etc--when the issue really is that they just need to update either php or composer. This is a real time-waster (and adds a ton of friction). It looks like all of the missing extensions and dependencies I was dealing with were all resolved by upgrading composer, so that is the step we should point users to.
- This also solves the "what happens when the person has not installed php or composer" issue?

**Comment by ViolanteCodes** [comment]
> @ViolanteCodes If any issues arise, should I commit directly to master? Because I created a new PR #3 where I removed the mentioned package.lock, so try switching to that branch and trying running `composer install` if the deprecation warnings still persist.

I saw the PR but still had the same issues (per above) so I didn't merge it in--please continue to open PRs for changes so that I can evaluate and test changes before merging in, and thank you.

**Comment by janondracek1** [comment]
@ViolanteCodes 
I will update the README.md and re-include composer.lock in the linked PR.
As far as I am aware, it is not possible to require one's version of composer.

**Comment by ViolanteCodes** [comment]
> @ViolanteCodes I will update the README.md and re-include composer.lock in the linked PR. As far as I am aware, it is not possible to require one's version of composer.

@janondracek1 we have already updated the language in the readme - sorry if that wasn't clear from commit history. I'm going to close this issue, thank you.
